src/main/java/org/folio/rest/impl/LoginAPI.java

```diff
@@ -199,31 +220,210 @@ private Future<String> fetchToken(JsonObject payload, String tenant,
       if (response.statusCode() != 200 && response.statusCode() != 201) {
         throw new RuntimeException("Got response " + response.statusCode() + " fetching token");
       }
-      String token = response.statusCode() == 200
-        ? response.getHeader(XOkapiHeaders.TOKEN)
-        : response.bodyAsJsonObject().getString("token");
-      if (token == null) {
-        throw new RuntimeException(String.format("Got response %s fetching token, but content is null",
-          response.statusCode()));
-      }
-      logger.debug("Got token " + token + " from authz");
-      return token;
+
+      return response.bodyAsJsonObject();
```

Comment:
Returning the Response rather than the token makes it easier to inspect the response code and handle that.